File: app/src/main/java/com/tobikster/medicreminder/di/AppComponent.java
package com.tobikster.medicreminder.di;

import android.app.Application;

import com.tobikster.medicreminder.MedicReminderApplication;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjectionModule;

@Singleton
@Component(modules = {
		AndroidInjectionModule.class,
        AppModule.class,
        RemindersListActivityModule.class
})
public interface AppComponent {
	@Component.Builder
	interface Builder {
		@BindsInstance
		Builder application(final Application application);
		AppComponent build();
	}
	void inject(final MedicReminderApplication medicReminderApplication);
}

File: app/src/main/java/com/tobikster/medicreminder/di/AppComponent.kt
package com.tobikster.medicreminder.di

import android.app.Application
import com.tobikster.medicreminder.MedicReminderApplication
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AndroidInjectionModule::class, AppModule::class, BuildersModule::class))
interface AppComponent {
	@Component.Builder
	interface Builder {
		@BindsInstance
		fun application(application: Application): Builder

		fun build(): AppComponent
	}

	fun inject(medicReminderApplication: MedicReminderApplication)
}
